### ISSUE TITLE
Pivot Cache is Invalid. Contains reference to an column with empty header

### ISSUE THREAD

**Issue by mwhite34** [EPPlusSoftware/EPPlus#567]
So I am using this excel file as a template:
https://docs.google.com/spreadsheets/d/e/2PACX-1vSH2SOMvSMOZVSAH5lWOTloVRZu1qhG6pSdn03HrsyKBGj1vPpwfxTR7lKdevp7RQ/pub?output=xlsx

The code is basically:
1. Load template
`using (var package = new ExcelPackage(new FileInfo(scheduledTemplateReport.TemplateFilePath)))`

2. Insert data into the detail tab of the report (starting 1 row below the column headers)
```
//code to load a collection to the spreadsheet. very nice
excelWorksheet.Cells[startCell].LoadFromArrays(dataCollection);
```

3. Update each pivot table's CacheDefinition.SourceRange to map to the data added in the detail tab
```
 foreach (TemplatePivot templatePivot in scheduledTemplateReport.TemplatePivots)
 {
    var pivotTable = package.Workbook.Worksheets[templatePivot.PivotSheetName].PivotTables[templatePivot.PivotName];

    pivotTable.CacheDefinition.SourceRange = package.Workbook.Worksheets[templatePivot.DataSheetName].Cells[templatePivot.DataRange];
 }
```

4. Save the new file with the updated data & pivot table ranges
`package.SaveAs(new FileInfo(scheduledTemplateReport.FilePath));`

No matter what I try, I always get an error when saving using EPPlus.

InvalidOperationException: Pivot Cache with id 5 is invalid . Contains reference to an column with empty header
   at OfficeOpenXml.Table.PivotTable.PivotTableCacheInternal.RefreshFields()
   at OfficeOpenXml.ExcelWorkbook.FixFieldNamesAndUpdateSharedItems(PivotTableCacheInternal cache, ExcelTable t, XmlNodeList fields)
   at OfficeOpenXml.ExcelWorkbook.SavePivotTableCaches()
   at OfficeOpenXml.ExcelWorkbook.Save()
   at OfficeOpenXml.ExcelPackage.Save()

I have debugged, manually input the data into the template, refreshed the tables and it will load correctly.

I have tested my logic w/out the pivot tables & confirmed the data fits perfectly under the column headers.

Not sure what else to try. I believe this code was working in the last 4.x version before I updated to 5+. Currently testing on 5.8.4.

**Comment by JanKallman** [comment]
EPPlus seems to have trouble with this template. I'll have a more closer look into this and get back to you.

**Comment by JanKallman** [comment]
I don't get the same error message as you do, but I have applied a fix for this test case...
```C#
  using (var package = OpenTemplatePackage("i567.xlsx"))
  {
      var wsSource = package.Workbook.Worksheets["Detail"];

      var dataCollection = new List<object[]>()
      { 
          new object[]{"Driver 1",1,2,"Fleet 1", "Manager 1",3,true,0,5,0 },
          new object[]{"Driver 2",3,4,"Fleet 2", "Manager 2", 5,true,0,8,0 }
      };
      //code to load a collection to the spreadsheet. very nice
      wsSource.Cells["A2"].LoadFromArrays(dataCollection);

      foreach(var ws in package.Workbook.Worksheets)
      {
          foreach(var pt in ws.PivotTables)
          {
              pt.CacheDefinition.SourceRange = wsSource.Cells["A1:J3"];
          }
      }
     SaveAndCleanup(package);
  }
```
Try it in the latest develop build... https://github.com/EPPlusSoftware/EPPlus/wiki/Using-our-Develop-Nuget-Feed
If you still get the same problem, attach a test case to reproduce your issue.

**Comment by mwhite34** [comment]
Thank you for the quick response!

I installed the latest version and I get a different error this time:
Value cannot be null.
Parameter name: value
   at System.String.StartsWith(String value, StringComparison comparisonType)
   at OfficeOpenXml.Table.PivotTable.PivotTableCacheInternal.RefreshFields()
   at OfficeOpenXml.ExcelWorkbook.FixFieldNamesAndUpdateSharedItems(PivotTableCacheInternal cache, ExcelTable t, XmlNodeList fields)
   at OfficeOpenXml.ExcelWorkbook.SavePivotTableCaches()
   at OfficeOpenXml.ExcelWorkbook.Save()
   at OfficeOpenXml.ExcelPackage.Save()

Here is a json of some test data (fleets & manager names changed):
https://drive.google.com/file/d/1MvC6V5PXHepemPx6I86xjadFeObOOzRx/view?usp=sharing

There are no values in the dataset as the procedure doesn't return them.

Is there anything that you're doing differently in the 
`SaveAndCleanup(package);`
or the 
`OpenTemplatePackage("i567.xlsx")`
methods?

**Comment by JanKallman** [comment]
Do you overwrite the headers in the source data table? This error suggests that one of the field names is null. The first row in the source data must contain the field names and they must not be null and must be be unique. I'll have a look if I can reproduce this behavior.

**Comment by JanKallman** [comment]
Doing this gives the same issue...
`wsSource.Cells["A1"].Value = null;`
Make sure you leave the header row intact when you write to the source range. I'll add a fix checking nulls here.

**Comment by mwhite34** [comment]
Ok yes I got this working now by re-writing my template.

The issue that I was having was regarding newline characters in the Detail tab headers. e.g. "/nDriver #", "Profit/nCenter", etc...
The workbook would error out with a pivot table issue.
I don't think those are being recognized correctly because when I re-made the workbook without the newline characters in the detail tab headers, it correctly saved.

Thanks for all of your help!

**Comment by JanKallman** [comment]
Fixes added in EPPlus 5.8.5
